src/core/xfa_fonts.js

```diff
@@ -198,11 +223,60 @@ function getXfaFontWidths(name) {
     return null;
   }
 
-  const { baseWidths, factors } = info;
+  const { baseWidths, baseMapping, factors } = info;
+  let rescaledBaseWidths;
   if (!factors) {
-    return baseWidths;
+    rescaledBaseWidths = baseWidths;
+  } else {
+    rescaledBaseWidths = baseWidths.map((w, i) => w * factors[i]);
   }
-  return baseWidths.map((w, i) => w * factors[i]);
+
+  let currentCode = -2;
+  let currentArray;
+
+  const newWidths = [];
+  for (const [code, i] of baseMapping
+    .map((c, j) => [c, j])
+    .sort(([c1], [c2]) => c1 - c2)) {
```

Comment:
This isn't exactly a wonder of readability (at least to me), any chance that it can be re-factored and/or simplified?